tests/e2e/protractor/sms/registration-by-sms.js

```diff
@@ -1,302 +0,0 @@
-      originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
-      jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
-    });
-
-    afterEach(() => utils.afterEach());
-    afterAll(() => {
-      jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
-    });
-
-    const checkItemSummary = async () => {
-      expect(await reportsPo.submitterName().getText()).toMatch(`Submitted by ${CAROL.name}`);
-      expect(await reportsPo.subjectName().getText()).toBe('Siobhan');
-      expect(await reportsPo.submitterPhone().getText()).toBe(CAROL.phone);
-      expect(await reportsPo.submitterPlace().getText()).toBe(BOB_PLACE.name);
-      expect(await reportsPo.detail().isDisplayed()).toBeTruthy();
-      expect(await reportsPo.detailStatus().isPresent()).toBeFalsy();
-    };
-
-    const checkAutoResponse = async (expectedDate) => {
-      expect(await reportsPo.taskTextByIndex(1)).toBe('Thank you '+ CAROL.name +' for registering Siobhan');
-      expect(await reportsPo.taskGatewayStatusByIndex(1).isDisplayed()).toBeTruthy();
-      expect(await reportsPo.taskRecipientByIndex(1).getText()).toBe(' to +64271234567');
-
-      expect(await reportsPo.taskTextByIndex(2)).toBe(`LMP ${expectedDate.locale('sw').format(dateFormatString)}`);
-      expect(await reportsPo.taskGatewayStatusByIndex(2).isDisplayed()).toBeTruthy();
-      expect(await reportsPo.taskRecipientByIndex(2).getText()).toBe(' to +64271234567');
-    };
-
-    const checkScheduledTask = async (childIndex, title, message) => {
-      expect(await reportsPo.scheduledTaskGroupByIndex(childIndex).element(by.css('h3')).getText()).toContain(title);
-      expect(await reportsPo.scheduledTaskMessageByIndex(childIndex).getText()).toBe(message);
-      expect(await reportsPo.scheduledTaskStateByIndex(childIndex).getText()).toBe('scheduled');
-      expect(await reportsPo.scheduledTaskRecipientByIndex(childIndex).getText()).toBe(' to +64271234567');
-    };
-
-    it('shows content', async () => {
-      await commonElements.goToReportsNative();
-      const firstReport = reportsPo.firstReport();
-      await helper.waitElementToBeClickable(firstReport);
-      await browser.wait(() => element(
-        by.cssContainingText(reportsPo.subject(firstReport).locator().value, 'Siobhan')
-      ).isPresent(), 10000);
-      await helper.clickElementNative(reportsPo.formName(firstReport));
-
-      // wait for content to load
-      await browser.wait(() => element(
-        by.cssContainingText('#reports-content .item-summary .phone', CAROL.phone)
-      ).isPresent(), 30000);
-
-      const expectedDate = await computeExpectedDate();
-
-      await checkItemSummary();
-      await checkAutoResponse(expectedDate);
-      await checkScheduledTask(1, 'ANC Reminders LMP:1', 'Visit 1 reminder for Siobhan');
-      await checkScheduledTask(2, 'ANC Reminders LMP:2', 'Visit 2 reminder for Siobhan');
-      await checkScheduledTask(3, 'ANC Reminders LMP:3', `LMP ${expectedDate.locale('sw').format(dateFormatString)}`);
```

Comment:
There is a lot of assertions here that do not see reflected in the new `registration-by-sms.wdio-spec.js`.  Is that intentional (is there already other tests for this stuff somewhere)? 

Honestly it feels like there are kind of 2 things going on here.  One is the basic registration via sms that  you have covered, but the other is the [registration transition](https://docs.communityhealthtoolkit.org/apps/reference/app-settings/transitions/#registration) that does a bunch of things based on configured [sms schedules](https://docs.communityhealthtoolkit.org/apps/tutorials/sms-schedules/).  This is the part that I am not really seeing in your new test file.  Maybe it should have its own file in `tests/e2e/default/transitions`?

Reply:
Thank you Josh for pointing this out, let me try to explain why I did the `wdio` test as it is.

In the protractor test three different outgoing messages are displayed because it is using a custom form, it generates the `ANC Reminders LMP:1`, `ANC Reminders LMP:2`, and `ANC Reminders LMP:3`. I am using the `"N"` form that is already part of the `standard` config, it generates only one message, the `Registration Followup`.

The method `checkScheduledTask()` asserts the title, the message content, the state of the message, and the phone number, for each message, and I am only asserting [the complete outgoing message section and the state](https://github.com/medic/cht-core/blob/refactor-registration-by-sms-test/tests/e2e/standard/sms/registration-by-sms.wdio-spec.js#L45-L46), I thought that the important part was that the outgoing message was generated and that it is as `scheduled`.

Do you think that I should also include asserts for the title and the content of the message?

Reply:
> In the protractor test three different outgoing messages are displayed because it is using a custom form

This is the part that I am unsure about.  To be honest, I don't have a lot of experience here with these SMS forms/schedules, but those three outgoing messages look like they are coming from the  `ANC Reminders LMP` schedules config (set up on line `171`. I may just be totally confused here, but I don't think that is quite the same as just a separate custom form.  From the docs that I linked above, it seems connected to the `registration` Sentinel transition (which tracks with the fact that this test file starts with `describe('registration transition', () => {...`.